### PR TITLE
Benchmark dirty trie

### PR DESCRIPTION
**Pull request by petr-hanzl**
This PR adds benchmark to test closing a Trie with high number of dirty nodes.

### CODE REVIEW

File: go/database/mpt/live_trie_test.go [hunk 1]
```diff
@@ -960,3 +960,34 @@ func BenchmarkValueInsertionInFileTrie(b *testing.B) {
...
+	// running this test for all configurations at once takes too much time
+	// hence we use only the latest one
+	config := S5ArchiveConfig
+	// largest nodes have ~512 bytes - we need to create a cache with 18 GB cache,
+	// hence we need ~ 18*2^30/2^9 = 18*2^21 ~ 38 million
+	cacheSize := 38_000_000
+	// number of accounts need to be ~1/3 smaller because we store leafs and branches
+	numAccounts := cacheSize - (cacheSize / 3)
+	dir := b.TempDir()
+	state, err := OpenGoFileState(dir, config, cacheSize)
```
Comment (HerbertJordan) [inline go/database/mpt/live_trie_test.go:974]: A benchmark targeting a state should not be in `live_trie_test.go` but in `state_test.go`

File: go/database/mpt/live_trie_test.go [hunk 2]
```diff
@@ -960,3 +960,34 @@ func BenchmarkValueInsertionInFileTrie(b *testing.B) {
 		})
 	}
 }
+
+func BenchmarkLiveTrie_WithHighNumberOfDirtyNodes_Flush(b *testing.B) {
+	// running this test for all configurations at once takes too much time
+	// hence we use only the latest one
+	config := S5ArchiveConfig
+	// largest nodes have ~512 bytes - we need to create a cache with 18 GB cache,
```
Comment (HerbertJordan) [inline go/database/mpt/live_trie_test.go:968]: I checked yesterday, while investigating memory issues reported by Jan: the largest nodes actually have 944 bytes.

File: go/database/mpt/live_trie_test.go [hunk 3]
```diff
@@ -960,3 +960,34 @@ func BenchmarkValueInsertionInFileTrie(b *testing.B) {
...
+	// hence we need ~ 18*2^30/2^9 = 18*2^21 ~ 38 million
+	cacheSize := 38_000_000
```
Comment (HerbertJordan) [inline go/database/mpt/live_trie_test.go:970]: As discussed offline, the problem size should be considerably reduced to make this a reasonable benchmark.

File: go/database/mpt/live_trie_test.go (hunk 2 above)
Comment (petr-hanzl) [reply go/database/mpt/live_trie_test.go:968]: It make sense. Grafana showed memory consumption of <30GiB. Thanks for the heads up!

File: go/database/mpt/live_trie_test.go (hunk 3 above)
Comment (petr-hanzl) [reply go/database/mpt/live_trie_test.go]: Unfortunately, I wasnt able to push this test under 30s.

File: go/database/mpt/state_test.go [hunk 4]
```diff
@@ -625,3 +625,44 @@ func TestState_parseCodes_ReadFailures(t *testing.T) {
...
+			if err != nil {
+				b.Fatalf("failed to create account: %v", err)
+			}
+			err = state.SetCode(addr, []byte{byte(j)})
+			if err != nil {
+				b.Fatalf("failed to set code: %v", err)
+			}
+		}
+
+		if err = state.Flush(); err != nil {
```
Comment (kjezek) [inline go/database/mpt/state_test.go:659]: should not we call only `Flush`() in the `b.N` for-loop?  At least for the bug with slow `Flush`(), I guess we are interested in performance of `Flush` only

**Comment by HerbertJordan** [comment]
This PR got superseded by #842

File: go/database/mpt/state_test.go (hunk 4 above)
Comment (HerbertJordan) [reply go/database/mpt/state_test.go:659]: Indeed, those operations in addition to the flush are very costly, dominating the execution time. The problem got fixed in #842 .

**Pull request closed without merging**